sdk/src/main/scala/com/horizen/SidechainMemoryPool.scala

```diff
@@ -61,6 +59,19 @@ class SidechainMemoryPool private(unconfirmed: MempoolMap, mempoolSettings: Memp
     unconfirmed.values.toSeq.sortWith(sortFunc).take(limit).map(tx => tx.getUnconfirmedTx())
   }
 
+  def takeWithWithdrawalBoxesLimit(allowedWithdrawalBoxes: Int): Iterable[SidechainTypes#SCBT] = {
+    val filteredTxs: JArrayList[SidechainTypes#SCBT] = new JArrayList[SidechainTypes#SCBT]()
+    var newWithdrawalBoxes = 0
+    take(size).foreach( tx => {
```

Comment:
take(size) is useless, we could do only unconfirmed.map or something similar

Reply:
The **take** function is sorting the transactions by fee rate that it's something that we want to keep. Please correct me if i'm wrong.